usr.bin/top/top.c

```diff
@@ -907,6 +908,20 @@ main(int argc, const char *argv[])
 				}
 				break;
 
+			    case CMD_grep: /* grep command name */
+				new_message(MT_standout,
+				    "Grep command name: ");
+				if (readline(tempbuf1, sizeof(tempbuf1), false) > 0)
+				{
+					if (tempbuf1[0] == '+' &&
+						tempbuf1[1] == '\0')
+						ps.command = NULL;
+					else if ((ps.command = strdup(tempbuf1)) == NULL)
+						quit(1);
+					clear_message();
```

Comment:
And this clear_message() needs to be outside the if, because we need to do it either way.

Reply:
ah, good catch